core/lb.c

```diff
@@ -563,15 +650,34 @@ _local__patch(struct oio_lb_pool_s *self,
 		.avoids = avoids,
 		.polled = (const oio_location_t *) polled,
 		.next_polled = polled,
+		.n_targets = count_targets,
 	};
 
+	for (int level = 1; level < 4; level++) {
+		g_datalist_init(&ctx.counters[level]);
+	}
+
 	guint count = 0;
 	for (gchar **ptarget = lb->targets; *ptarget; ++ptarget) {
 		gboolean done = _local_target__is_satisfied(lb, *ptarget, &ctx);
-		guint mask_shift = 0u;
-		while (!done && mask_shift <= lb->location_mask_max_shift) {
-			done = _local_target__poll(lb, *ptarget, mask_shift, &ctx);
-			mask_shift += 8u;  // Degrade mask by 8 bits (two hex digit)
+		/* In normal mode (resp. nearby mode), bit shifts start high
+		 * (resp. low), because we want services with different
+		 * (resp. equal) most significant bits. Then we reduce (resp. increase)
+		 * the shifting so we compare more (resp. less) bits of the locations,
+		 * and thus have more chances to find differences (resp. similarities)
+		 * between service locations. */
+		if (lb->nearby_mode) {
+			for (gint bit_shift = lb->min_shift;
+					!done && bit_shift <= lb->initial_shift;
+					bit_shift += 16) {
+				done = _local_target__poll(lb, *ptarget, bit_shift, &ctx);
+			}
+		} else {
+			for (gint bit_shift = lb->initial_shift;
+					!done && bit_shift >= lb->min_shift;
+					bit_shift -= 16) {
```

Comment:
Isn't here a risk of underflow if `lb->min_shift` has been configured to less than `16`?